lenskit-core/src/main/java/org/grouplens/lenskit/data/event/Like.java

```diff
@@ -20,14 +20,120 @@
  */
 package org.grouplens.lenskit.data.event;
 
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import java.io.Serializable;
+
 /**
  * Generic interface representing a unary interaction.  This provides a basic, generic representation
  * of likes, clicks, purchases, and other unary interactions.  Applications may want to implement
  * their own events to represent such interactions.
+ * <p>
+ * This class can be created with {@link LikeBuilder} or the {@link #create(long, long, long)} method.
+ * </p>
  *
  * @since 2.2
  * @see Events#like(long, long)
  * @see Events#like(long, long, long)
  */
-public interface Like extends Event {
+public final class Like implements Event, Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private final long userId;
+    private final long itemId;
+    private final long timestamp;
+
+    Like(long uid, long iid, long ts) {
```

Comment:
If memory serves, the argument against using constructors has to do with "what happens when we need to change the object and how its constructed" events should be very simple POJOs. Any significant change to how we build a like is probably precipitated by a change in what is a like, which would be a breaking change anyways. so I guess Im not sure I buy that argument in this case.